rasa_core/channels/botframework.py

```diff
@@ -0,0 +1,169 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import json
+import logging
+from requests import post
+import datetime
+
+from flask import Blueprint, request, jsonify
+
+from rasa_core.channels.channel import UserMessage, OutputChannel
+from rasa_core.channels.rest import HttpInputComponent
+
+logger = logging.getLogger(__name__)
+
+
+class BotFramework(OutputChannel):
+    """A Microsoft Bot Framework communication channel."""
+
+    token_expiration_date = datetime.datetime.now()
+    headers = None
+
+    def __init__(self, bf_id, bf_secret, conversation, bot_id,
+                 service_url):
+        # type: (Text, Text, Dict[Text]) -> None
+
+        self.bf_id = bf_id
+        self.bf_secret = bf_secret
+        self.conversation = conversation
+        self.global_uri = "%sv3/" % service_url
+        self.bot_id = bot_id
+
+    def get_headers(self):
+        if BotFramework.token_expiration_date < datetime.datetime.now():
+            microsoft_oauth2_url = 'https://login.microsoftonline.com'
```

Comment:
can we make these two constants at the top of the file please?  (module level)

```
MICROSOFT_OAUTH2_URL = 'https://login.microsoftonline.com'

MICROSOFT_OAUTH_PATH = 'botframework.com/oauth2/v2.0/token'
```

Reply:
even `MICROSOFT_OAUTH2_PATH` no ?